tests/test_data.py

```diff
@@ -0,0 +1,69 @@
+import io
+import json
+import os
+import tempfile
+
+import pytest
+
+import rasa.data as data
+
+
+def test_get_core_directory(project):
+    data_dir = os.path.join(project, "data")
+    core_directory = data.get_core_directory(data_dir)
+
+    assert os.listdir(core_directory) == ["stories.md"]
+
+
+def test_get_nlu_directory(project):
+    data_dir = os.path.join(project, "data")
+    nlu_directory = data.get_nlu_directory(data_dir)
+
+    assert os.listdir(nlu_directory) == ["nlu.md"]
+
+
+def test_get_core_nlu_directories(project):
+    data_dir = os.path.join(project, "data")
+    core_directory, nlu_directory = data.get_core_nlu_directories(data_dir)
+
+    assert os.listdir(core_directory) == ["stories.md"]
+    assert os.listdir(nlu_directory) == ["nlu.md"]
+
+
+@pytest.mark.parametrize("line", [
+    "##intent:aintent",
+    "##synonym: synonym",
+    "##regex:a_regex",
+    " ##lookup:additional"])
+def test_contains_nlu_pattern(line):
+    assert data._contains_nlu_pattern(line)
+
+
+def test_is_nlu_file_with_json():
+    test = {"rasa_nlu_data": {"lookup_tables": [
+        {"name": "plates", "elements": ["beans", "rice", "tacos", "cheese"]}]}}
+
+    directory = tempfile.mkdtemp()
+    file = os.path.join(directory, "test.json")
+    with io.open(file, "w") as f:
+        f.write(json.dumps(test))
+
+    assert data._is_nlu_file(file)
+
+
+def test_is_not_nlu_file_with_json():
+    directory = tempfile.mkdtemp()
+    file = os.path.join(directory, "test.json")
+    with io.open(file, "w") as f:
```

Comment:
we shouldn't use `io.open` anymore, in > py3.5 there is no difference to just `open`